go-controller/pkg/ovn/common.go

```diff
@@ -257,18 +257,32 @@ func (oc *Controller) getNetworkInfoFromOvnAnnotation(ovnAnnotation string, netw
 	return ovnExtraAnnotationMap[networkName]
 }
 
-func (oc *Controller) getNetworkNamesFromPodAnnotations(podAnnotations map[string]string) []string {
-	switchesAnnotation, hasExtraSwitches := podAnnotations["switches"]
+func (oc *Controller) getNetworkNamesFromPodAnnotations(podAnnotations map[string]string, namespace string) []string {
+	networksAnnotation, hasExtraNetworks := podAnnotations["k8s.v1.cni.cncf.io/networks"]
 
-	if hasExtraSwitches {
-		var switches []string
-		err := json.Unmarshal([]byte(switchesAnnotation), &switches)
+	if hasExtraNetworks {
+		var networks []map[string]string
+		err := json.Unmarshal([]byte(networksAnnotation), &networks)
 
 		if err != nil {
-			logrus.Errorf("Error in json unmarshaling switches annotation  (%v)", err)
+			logrus.Errorf("Error in json unmarshaling networks annotation  (%v)", err)
 			return nil
 		}
-		return switches
+
+		var networkNames []string
+		for _, network := range networks {
+			networkAttachmentName := network["name"]
+			config, err := oc.kube.GetConfigFromNetworkAttachment(namespace, networkAttachmentName)
+
+			if err != nil {
+				logrus.Errorf("Error getting the config of network attachment %s, (%v)", networkAttachmentName, err)
+				return nil
+			}
+			if config["type"] == "ovn-k8s-cni-overlay" {
+				networkNames = append(networkNames, config["name"].(string))
+			}
```

Comment:
maybe submit an info/warning on the "else"? if we found the same network name for a different cni, may be an issue?

Reply:
Each cni can use the networkAttachment.spec.config.name attribute for different purpose. As far as I know there is no standard about it. So, it is ok if different cni has the same spec.config.name.
Maybe to avoid confusion we should introduce a new key instead of 'name'. Maybe switch_name? What do you think?

Reply:
I think the name needs to be the same as the networkAttachment CRD and create a new variable, maybe just `switch`.

For reference on the cni bridge plugin
```
{
	"name": "mynet",
	"type": "bridge",
	"bridge": "mynet0", #bridge name
        ...
}
```
https://github.com/containernetworking/plugins/tree/master/plugins/main/bridge

Reply:
according to the spec, every CNI can add whatever info they want. however, to prevent collisions with future enhancements to the spec, they ask the names to be of the form: xxx.fieldName
where "xxx" is a unique identified of the CNI that added this field

Reply:
please see [this](https://github.com/kubernetes/community/pull/2665/files#diff-b7df51911eac0f70e97ab917caf1156bR320)

Reply:
Good one!
so maybe `ovn.switch` or something like that?